### PR TITLE
cpu/stm32f0: Add support for stm32f030cc CPU

### PR DESCRIPTION
**Pull request by niziak**

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@niziak thanks for your PR (again). Please don't forget to always add a short description to your contribution. The PR template should've suggested a structure. Which board uses this MCU?

**Comment by PeterKietzmann** [comment]
(relates to #7958)

**Comment by niziak** [comment]
This CPU is used on custom board (simple GSM metering device, USART0 is console, USART2 and USART3 is connected to GSM modem module).

**Comment by PeterKietzmann** [comment]
@vincent-d did you run some basic tests?

**Comment by vincent-d** [comment]
@PeterKietzmann no, I just checked that all the definitions were good

**Comment by PeterKietzmann** [comment]
@vincent-d do you have a device to test this?

**Comment by vincent-d** [comment]
@PeterKietzmann I don't. The best we could do is check that it builds with a "fake" board, I guess.

**Comment by aabadie** [comment]
Changes look good here. CI was green for quite some time so I retriggered Murdock.

Can we merge this PR @vincent-d @PeterKietzmann ?

**Comment by PeterKietzmann** [comment]
To be honest, I don't know. The code looks ok but it is not used anywhere. That means, it is dead code.

**Comment by PeterKietzmann** [comment]
@smlng, @aabadie what's your opinion here, merge or close?

File: cpu/stm32f0/include/cpu_conf.h [hunk 1]
```diff
@@ -55,7 +56,8 @@ extern "C" {
  *
  * @{
  */
-#if defined(CPU_LINE_STM32F091xC) || defined(CPU_LINE_STM32F072xB)
+#if defined(CPU_LINE_STM32F091xC) || defined(CPU_LINE_STM32F072xB) \
```
Comment (smlng) [inline cpu/stm32f0/include/cpu_conf.h]: this should go in its own commit

File: cpu/stm32f0/vectors.c [hunk 2]
```diff
@@ -218,5 +219,24 @@ ISR_VECTOR(1) const isr_t vector_cpu[CPU_IRQ_NUMOF] = {
     [28] = isr_usart2,               /* [28] USART2 global Interrupt & EXTI Line26 Interrupt (USART2 wakeup) */
     [29] = isr_usart3_8,             /* [29] USART3 to USART8 global Interrupts */
     [30] = isr_cec_can,              /* [30] CEC and CAN global Interrupts & EXTI Line27 Interrupt */
+#elif defined(CPU_LINE_STM32F030xC)
```
Comment (smlng) [inline cpu/stm32f0/vectors.c]: this also needs its own commit

**Comment by benpicco** [comment]
Well we also don't have a board with a samd51 MCU in RIOT and yet I maintain it - because we have a custom board with that MCU.

It's also just a same54 without the Ethernet (and some other features) for which a dev board exists.

I don't think we need boards for all siblings of a MCU family in upstream RIOT as long as there is one member with a dev board and they are similar enough.

**Comment by PeterKietzmann** [comment]
So @smlng and @benpicco are prone to merge this – then it's fine with me. @niziak would you please address @smlng's comments?

File: cpu/stm32f0/include/vendor/stm32f030xc.h [hunk 3]
```diff
@@ -0,0 +1,5852 @@
...
+  __IO uint32_t CNT;          /*!< TIM counter register,                Address offset: 0x24 */
+  __IO uint32_t PSC;          /*!< TIM prescaler register,              Address offset: 0x28 */
+  __IO uint32_t ARR;          /*!< TIM auto-reload register,            Address offset: 0x2C */
+  __IO uint32_t RCR;             /*!< TIM  repetition counter register,            Address offset: 0x30 */
+#if 0
+  __IO uint32_t CCR1;         /*!< TIM capture/compare register 1,      Address offset: 0x34 */    
+  __IO uint32_t CCR2;         /*!< TIM capture/compare register 2,      Address offset: 0x38 */    
+  __IO uint32_t CCR3;         /*!< TIM capture/compare register 3,      Address offset: 0x3C */
+  __IO uint32_t CCR4;         /*!< TIM capture/compare register 4,      Address offset: 0x40 */
+#endif
```
Comment (benpicco) [inline cpu/stm32f0/include/vendor/stm32f030xc.h]: Just drop this.

**Comment by aabadie** [comment]
This one is all green. @benpicco, ok to merge ?
